File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/user/dto/UserResponse.java
package com.portfolio.auctionmarket.domain.user.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.portfolio.auctionmarket.domain.sellers.entity.Seller;
import com.portfolio.auctionmarket.domain.user.entity.Role;
import com.portfolio.auctionmarket.domain.sellers.entity.SellerStatus;
import com.portfolio.auctionmarket.domain.user.entity.User;
import com.portfolio.auctionmarket.domain.user.entity.UserStatus;
import com.portfolio.auctionmarket.global.util.MaskingUtil;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserResponse {
    private Long userId;
    private String email;
    private String username;
    private String phone;
    private String nickname;
    private String baseAddress;
    private String detailAddress;
    private Role role;
    private SellerStatus sellerStatus;
    private UserStatus userStatus;
    private Long point;
    private String avgRating;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdAt;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime updatedAt;

    public static UserResponse from(User user){
        String formatPhone = MaskingUtil.formatPhone(user.getPhone());
        Seller seller = user.getSeller();
        SellerStatus sellerStatus = (seller != null) ? seller.getStatus() : SellerStatus.NONE;

        return UserResponse.builder()
                .userId(user.getUserId())
                .email(user.getEmail())
                .username(user.getUsername())
                .phone(formatPhone)
                .nickname(user.getNickname())
                .baseAddress(user.getBaseAddress())
                .detailAddress(user.getDetailAddress())
                .role(user.getRole())
                .userStatus(user.getStatus())
                .point(user.getPoint())
                .avgRating(String.format("%.1f", user.getAvgRating()))
                .sellerStatus(sellerStatus)
                .createdAt(user.getCreatedAt())
                .updatedAt(user.getUpdatedAt())
                .build();
    }
}

File: AuctionMarket/auction-app-user/src/main/kotlin/org/example/auctioncommon/domain/user/dto/UserResponse.kt
package org.example.auctioncommon.domain.user.dto

import com.fasterxml.jackson.annotation.JsonFormat
import org.example.auctioncommon.domain.seller.entity.Seller
import org.example.auctioncommon.domain.seller.entity.SellerStatus
import org.example.auctioncommon.domain.user.entity.Role
import org.example.auctioncommon.domain.user.entity.User
import org.example.auctioncommon.domain.user.entity.UserStatus
import org.example.auctioncommon.global.util.MaskingUtil
import java.time.LocalDateTime
import kotlin.Long

data class UserResponse(
    var userId: Long? = null,
    var email: String,
    var username: String,
    var phone: String,
    var nickname: String,
    var baseAddress: String,
    var detailAddress: String? = null,
    var role: Role,
    var sellerStatus: SellerStatus? = null,
    var userStatus: UserStatus,
    var point: Long,
    var avgRating: Double,

    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val createdAt: LocalDateTime? = null,
    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val updatedAt: LocalDateTime? = null
)

fun User.toDto() : UserResponse {
    val formatPhone = MaskingUtil.formatPhone(phone)
    val seller: Seller? = seller
    val sellerStatus: SellerStatus? = if (seller != null) seller.status else SellerStatus.NONE

    return UserResponse(
        userId = this.userId,
        email = this.email,
        username = this.username,
        phone = formatPhone.toString(),
        nickname = this.nickname,
        baseAddress = this.baseAddress,
        detailAddress = this.detailAddress,
        role = this.role,
        sellerStatus = sellerStatus,
        userStatus = this.status,
        point = this.point,
        avgRating = this.avgRating,
        createdAt = this.createdAt,
        updatedAt = this.updatedAt
    )
}